guppylang/hugr_builder/hugr.py

```diff
@@ -746,16 +810,16 @@ def to_raw(self) -> raw.RawHugr:
         all_nodes = self.nodes()
         root_node = next(all_nodes)
         for n in all_nodes:
-            if isinstance(n.op, ops.Input):
+            if isinstance(n.op.root, ops.Input):
                 input_nodes.append(n)
-            elif isinstance(n.op, ops.Output):
+            elif isinstance(n.op.root, ops.Output):
                 output_nodes.append(n)
             elif (
-                isinstance(n.op, ops.DataflowBlock)
+                isinstance(n.op.root, ops.DataflowBlock)
```

Comment:
> Is this to put the entry block in first?

Yes!

> Possibly you avoid generating such an entry block yourself, in which case this is probably worth a comment

Yeah, Guppy will never generate jumps to the entry BB since loops always create a new BB for the test and jump back to that. I'll add a comment.

> What if there's an infinite loop, so the code after it is unreachable (and really does have no edges in) - can we generate such a CFG?

We can't generate these right now. Guppy doesn't know that the statement `True` is always true, so it will always generate a branch for the case that it's false. But once Guppy is smarter about this, it will indeed become a problem! I'll move the case checking for exits above the one checking for entries, this should fix the issue.